src/python/pants/util/objects.py

```diff
@@ -537,7 +536,9 @@ def satisfied_by_type(self, obj_type):
     return issubclass(obj_type, self._types)
 
 
-_string_type_constraint = SubclassesOf(binary_type, text_type)
+# TODO(#6071): We should in general not allow bytes _and_ str. The whole point of Py3 was to decide
```

Comment:
It would be good to open a more specific ticket for this probably.

Reply:
I'm planning on addressing this pretty soon, which is why I didn't open a new ticket. Also, #6071 already has several checklist items to tighten up expectations around `str` vs. `bytes`.